app/controllers/test_data_controller.rb

```diff
@@ -0,0 +1,421 @@
+class TestDataController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :render_not_found
+
+  def create_test_lettings_log
+    return render_not_found unless FeatureToggle.create_test_logs_enabled?
+
+    log = FactoryBot.create(:lettings_log, :completed, assigned_to: current_user, ppostcode_full: "SW1A 1AA")
+    redirect_to lettings_log_path(log)
+  end
+
+  def create_setup_test_lettings_log
+    return render_not_found unless FeatureToggle.create_test_logs_enabled?
+
+    log = FactoryBot.create(:lettings_log, :setup_completed, assigned_to: current_user)
+    redirect_to lettings_log_path(log)
+  end
+
+  def create_2024_test_lettings_bulk_upload
+    return render_not_found unless FeatureToggle.create_test_logs_enabled?
+
+    file = Tempfile.new("test_lettings_log.csv")
+    header = ["Field number"] + (1..130).to_a
+    file.write("#{header.join(',')}\n")
+    log = FactoryBot.create(:lettings_log, :completed, assigned_to: current_user, ppostcode_full: "SW1A 1AA")
+    file.write("#{([nil] + to_2024_lettings_row(log)).flatten.join(',')}\n")
+    file.rewind
+    send_file file.path, type: "text/csv",
+                         filename: "test_lettings_log.csv",
+                         disposition: "attachment",
+                         after_send: lambda {
+                                       file.close
+                                       file.unlink
+                                     }
+  end
+
+  def create_test_sales_log
+    return render_not_found unless FeatureToggle.create_test_logs_enabled?
+
+    log = FactoryBot.create(:sales_log, :completed, assigned_to: current_user)
+    redirect_to sales_log_path(log)
+  end
+
+  def create_setup_test_sales_log
+    return render_not_found unless FeatureToggle.create_test_logs_enabled?
+
+    log = FactoryBot.create(:sales_log, :shared_ownership_setup_complete, assigned_to: current_user)
+    redirect_to sales_log_path(log)
+  end
+
+  def create_2024_test_sales_bulk_upload
+    return render_not_found unless FeatureToggle.create_test_logs_enabled?
+
+    file = Tempfile.new("test_sales_log.csv")
+    header = ["Field number"] + (1..131).to_a
+    file.write("#{header.join(',')}\n")
+    log = FactoryBot.create(:sales_log, :completed, assigned_to: current_user, value: 180_000, deposit: 150_000)
+    file.write("#{([nil] + to_2024_sales_row(log)).flatten.join(',')}\n")
+    file.rewind
+    send_file file.path, type: "text/csv",
+                         filename: "test_sales_log.csv",
+                         disposition: "attachment",
+                         after_send: lambda {
+                                       file.close
+                                       file.unlink
+                                     }
+  end
+
+private
+
+  def to_2024_lettings_row(log)
```

Comment:
Hm, yeah, probably, I tried to avoid requiring files from tests and they weren't getting autoloaded. Realistically it probably is fine, but it doesn't feel right. 
Any ideas other than just requiring the test helper files?

Reply:
Could they live somewhere around here (probably not this file) and then be required by the tests?

Reply:
I moved them to helpers, does that look ok?

Reply:
I think it is (?) it's very weird, but in the file changes it shows up as a rename, so the files in `spec/support/bullk_upload` no longer exist, at least when I pull it down

Reply:
Huh, strange. Cool, lets get it in.